### PR TITLE
[ADD] purchase_stock_picking_invoice_link: linking purchases, picking…

### PR DESCRIPTION
**Pull request by cubells**
…s and invoices

This module solves a missing feature in stock_picking_invoice_link: link purchases, pickings and invoices

@pedrobaeza @js-landoo @omar7r  can you review?

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by AdriaGForgeFlow — APPROVED** [review]
Code review :+1:

**Review by LoisRForgeFlow — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I have aborted the merge because my request is still not solved:

https://github.com/OCA/stock-logistics-workflow/pull/500#pullrequestreview-203812994

If you use the purchase order selector, which activates an onchange, moves are not linked.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
See previous comment

File: purchase_stock_picking_invoice_link/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    def _prepare_invoice_line_from_po_line(self, line):
+        vals = super()._prepare_invoice_line_from_po_line(line)
+        moves = self.env['stock.move'].search([
+            ('purchase_line_id', '=', line.id),
+        ])
+        if moves:
+            move_ids = moves._get_moves()
+            vals['move_line_ids'] = [(6, 0, move_ids.ids)]
+            pickings = move_ids.mapped('picking_id')
+            pickings.invoice_ids = [(4, self.id)]
```
Comment (omar7r) [inline purchase_stock_picking_invoice_link/models/account_invoice.py:20]: Here, we should do the same as below, loop in pickings for writing invoice_ids, it fails with partial pickings for the same line.

**Comment by rousseldenis** [comment]
@cubells

File: purchase_stock_picking_invoice_link/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return vals
+
+    @api.model
+    def create(self, values):
+        pickings = self.env['stock.picking']
+        result = []
+        if 'invoice_line_ids' not in values:
+            return super().create(values)
+        for item in values.get('invoice_line_ids'):
+            if not item[2].get('purchase_line_id'):
```
Comment (omar7r) [inline purchase_stock_picking_invoice_link/models/account_invoice.py:30]: invoice_line_ids can come formatted as [(6, 0, ids)] in this case, this line fails

**Comment by rousseldenis** [comment]
@pedrobaeza Is your review still applicable?

**Comment by cubells** [comment]
@rousseldenis Yes, it is comment of  @pedrobaeza applicable

**Comment by pedrobaeza** [comment]
Closing as not finished. 13.0 version in #883

**Pull request closed without merging**